src/vm/wren_value.c

```diff
@@ -1021,25 +1021,17 @@ static void markString(WrenVM* vm, ObjString* string)
   vm->bytesAllocated += sizeof(ObjString) + string->length + 1;
 }
 
-static void markUpvalue(WrenVM* vm, ObjUpvalue* upvalue)
+static void markUpvalue(WrenVM* vm, ObjUpvalue* upvalue, GCState* state)
 {
   // Mark the closed-over object (in case it is closed).
-  wrenMarkValue(vm, upvalue->closed);
+  wrenMarkValue(vm, upvalue->closed, state);
 
   // Keep track of how much memory is still in use.
   vm->bytesAllocated += sizeof(ObjUpvalue);
 }
 
-void wrenMarkObj(WrenVM* vm, Obj* obj)
+static void expandMark(WrenVM* vm, Obj* obj, GCState* state)
 {
-  if (obj == NULL) return;
-
-  // Stop if the object is already marked so we don't get stuck in a cycle.
-  if (obj->marked) return;
-
-  // It's been reached.
-  obj->marked = true;
-
 #if WREN_DEBUG_TRACE_MEMORY
   static int indent = 0;
```

Comment:
Seen as we don't visit each object recursively any more this indented trace output doesn't really work. I can't really work out how to solve this at the moment i'm suggesting just removing the indentation completely from the tracing.